### PR TITLE
Show chapter title on seek bar

### LINKED ISSUE #2829: [Feature Request]: Show chapter title on seek bar
### Guidelines

- [X] I have searched the issue tracker for [open](https://github.com/FreeTubeApp/FreeTube/issues?q=is%3Aopen+is%3Aissue) and [closed](https://github.com/FreeTubeApp/FreeTube/issues?q=is%3Aissue+is%3Aclosed) issues that are similar to the feature request I want to file, without success.
- [X] I have searched the [documentation](https://docs.freetubeapp.io/) for information that matches the description of the feature request I want to file, without success.
- [X] This issue contains only one feature request.

### Problem Description

The chapters are not visible on the timeline 
they appear in a window below the video but not on the main timeline 
Screenshot 2022-11-04 at 11 45 13 PM
Screenshot 2022-11-04 at 11 45 21 PM

### Proposed Solution

It should show the chapter title when hovering over timeline

### Issue Labels

ease of use improvement, improvement to existing feature

### PR DESCRIPTION
**Pull request by Benjababe**
# Show Chapter Title and Sponsorblock segment on Seek Bar

## Pull Request Type

- [x] Feature Implementation

## Related issue

Closes #2829 

## Description
Includes the current chapter and sponsorblock segment hovered with the time in the seeker info, only affects videos with chapters or sponsorblock segments.
Jitters sometimes when the tooltips exceeds the player as it's being moved back into the player's boundaries

Updated `bottom` css attribute for the thumbnail to update as user hovers over the progress bar, positioning according to how many tooltips are shown

## Screenshots 

[Before](https://github.com/FreeTubeApp/FreeTube/assets/19561469/a65f5cc0-11dd-43e6-bdee-a3d58a618ef5)

https://github.com/FreeTubeApp/FreeTube/assets/19561469/c665aa82-f0fd-4ec6-b4a0-4dcfc2840c42

https://github.com/FreeTubeApp/FreeTube/assets/19561469/ddb113fa-7bc4-4c5b-be8d-5ece4ee9369f

## Testing 
Tried on [this](https://www.youtube.com/watch?v=Q9ejVtGh2Vw) and [this](https://www.youtube.com/watch?v=oWXRkO9pmHM) as well as a couple of random recommended videos.

## Desktop

- **OS:** Windows 10
- **OS Version:** 22H2
- **FreeTube version:**

### CODE REVIEW

**Review by PikachuEXE — CHANGES_REQUESTED** [review]
1. Missing final new line (already set in `.editorconfig`)
2. There are a lot of changes in `src/renderer/videoJS.css` but many changes are formatting changes. Possible to revert those for easier review?

**Comment by absidue** [comment]
I agree with pikachu please revert all the unrelated formatting changes. I don't know if you have your editor set to automatically format the files or something, but unrelated formatting changes seem to be a common pattern across all of your pull requests.

**Comment by Benjababe** [comment]
> Comparing the first and second video, it looks like the progress bar no longer stretches the full width of the player.

Thanks for spotting it, I had made a small mistake with the styling

**Review by efb4f5ff-1298-471a-8973-3d47447115dc — APPROVED** [review]
LGTM

**Comment by absidue** [comment]
I'm really not looking forward to trying to port this to shaka (will probably require even more code there, as you'll have to fight against the javascript that scales and positions their built-in thumbnail display, every time the mouse moves) :(

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Should this wait until we've migrated?

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
> would you prefer to not have it at all, until someone creates a pull request to add it to shaka, after we've got the shaka stuff merged in?

I prefer this tbh

File: src/renderer/components/ft-video-player/ft-video-player.js [hunk 1]
```diff
@@ -602,12 +604,39 @@ export default defineComponent({
...
+          })
+
+          if (visibleChildren.length === 0) {
+            return
+          }
+
+          const thumbnail = this.player.el().querySelector('.vjs-vtt-thumbnail-display')
+          if (!thumbnail) {
+            return
+          }
```
Comment (PikachuEXE) [inline src/renderer/components/ft-video-player/ft-video-player.js:626]: Too long
```suggestion
          if (!thumbnail) { return }
```

File: src/renderer/components/ft-video-player/ft-video-player.js [hunk 2]
```diff
@@ -765,6 +794,56 @@ export default defineComponent({
...
+          // If tooltip extends outside of the video player
+          // Adjust CSS right property to fit the tooltip within the player
+          if (sbTooltipLeftX < playerLeft) {
+            calculatedRight -= (playerLeft - sbTooltipLeftX)
+          } else if (sbTooltipRightX > playerRight) {
+            calculatedRight += (sbTooltipRightX - playerRight)
+          }
+
+          sponsorBlockTooltip.style.right = `${calculatedRight}px`
+        }
```
Comment (PikachuEXE) [inline src/renderer/components/ft-video-player/ft-video-player.js:843]: Unnecessary nesting IMO
```suggestion
        if (noMarkerHovered) {
          sponsorBlockTooltip.style.visibility = 'hidden'
          return
        }
        
        const { left: mouseDisplayLeft } = mouseDisplay.getBoundingClientRect()
        const { left: playerLeft, right: playerRight } = this.player.el().getBoundingClientRect()
        const { width: sbTooltipWidth } = sponsorBlockTooltip.getBoundingClientRect()

        let calculatedRight = -(sbTooltipWidth / 2)

        // Gets left and right boundaries of the chapter tooltip
        // +1 for rightX to include width of the mouse display
        const sbTooltipLeftX = mouseDisplayLeft - (sbTooltipWidth / 2)
        const sbTooltipRightX = mouseDisplayLeft + 1 + (sbTooltipWidth / 2)

        // If tooltip extends outside of the video player
        // Adjust CSS right property to fit the tooltip within the player
        if (sbTooltipLeftX < playerLeft) {
          calculatedRight -= (playerLeft - sbTooltipLeftX)
        } else if (sbTooltipRightX > playerRight) {
          calculatedRight += (sbTooltipRightX - playerRight)
        }

        sponsorBlockTooltip.style.right = `${calculatedRight}px`
```

File: src/renderer/components/ft-video-player/ft-video-player.js [hunk 3]
```diff
@@ -602,12 +604,39 @@ export default defineComponent({
...
+          const children = Array.from(mouseDisplay.children)
+          const visibleChildren = children.filter((child) => {
+            const style = window.getComputedStyle(child)
+            const visibility = style.getPropertyValue('visibility')
+            return visibility === 'visible'
+          })
+
+          if (visibleChildren.length === 0) {
+            return
+          }
```
Comment (PikachuEXE) [inline src/renderer/components/ft-video-player/ft-video-player.js:621]: I am not sure if the blank line should be kept, up to you
```suggestion
          if (visibleChildren.length === 0) { return }
```

**Review by PikachuEXE — CHANGES_REQUESTED** [review]

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
@absidue do you recommend putting this on hold?

**Comment by absidue** [comment]
Yes, especially as I see no way of implementing this with shaka-player, without modifying shaka-player itself. Without modifying shaka-player, you will be in a constant battle against it's JavaScript, because the entire thumbnail display is managed through JavaScript, even the positioning, so that is usable without their CSS (to allow people to use custom themes for the player).

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Will revisit this one after the migration.

**Pull request closed without merging**